src/plugins/xfer/xfer-dcc.c

```diff
@@ -368,9 +368,14 @@ xfer_dcc_recv_file_child (struct t_xfer *xfer)
         ready = poll (&poll_fd, 1, -1);
         if (ready <= 0)
         {
-            xfer_network_write_pipe (xfer, XFER_STATUS_FAILED,
-                                     XFER_ERROR_RECV_BLOCK);
-            return;
+            if ((errno == EINTR) || (errno == EAGAIN))
```

Comment:
This isn't checking the result of the write() but the result of the poll() above. `SIGWINCH` caused that poll() to return `EINTR` making it think the poll() failed, while it actually did not.